domain-classes-generator/src/main/scala/flatgraph/codegen/CodeSnippets.scala

```diff
@@ -126,10 +126,18 @@ object CodeSnippets {
          |/**
          |  * Traverse to nodes where $nameCamelCase matches one of the elements in `values` exactly.
          |  * */
-         |def ${nameCamelCase}Exact(values: $baseType*): Iterator[NodeType] =
-         |  if(values.length == 1) ${nameCamelCase}Exact(values.head) else {
-         |  val valueSet = values.toSet
-         |  traversal.filter{item => valueSet.contains(item.$nameCamelCase)}
+         |def ${nameCamelCase}Exact(values: $baseType*): Iterator[NodeType] = {
+         |  if(values.length == 1) return ${nameCamelCase}Exact(values.head)
+         |  traversal match {
+         |      case init: flatgraph.misc.InitNodeIterator[flatgraph.GNode @unchecked] if init.isVirgin && init.hasNext =>
+         |      val someNode = init.next
+         |      values.iterator.flatMap{  v =>
+         |        flatgraph.Accessors.getWithInverseIndex(someNode.graph, someNode.nodeKind,  $propertyId, v).asInstanceOf[Iterator[NodeType]]
+         |      }
+         |      case _ =>
+         |      val valueSet = values.toSet
+         |      traversal.filter{item => valueSet.contains(item.$nameCamelCase)}
+         |    }
```

Comment:
```suggestion
         |  traversal match {
         |    case init: flatgraph.misc.InitNodeIterator[flatgraph.GNode @unchecked] if init.isVirgin && init.hasNext =>
         |      val someNode = init.next
         |      values.iterator.flatMap { value =>
         |        flatgraph.Accessors.getWithInverseIndex(someNode.graph, someNode.nodeKind,  $propertyId, value).asInstanceOf[Iterator[NodeType]]
         |      }
         |    case _ =>
         |      val valueSet = values.toSet
         |      traversal.filter{item => valueSet.contains(item.$nameCamelCase)}
         |    }
```

^ just indentation and one minor name fix